### ISSUE TITLE
view.setCenter() with coordinates as strings causes inconsistent behaviour

### ISSUE THREAD

**Issue by derekwatling** [openlayers/ol3#6016]
When calling view.setCenter() with the coordinate array containing "numbers" as strings the map will change position to the correct center, but then some functinallity works while other functionality returns the view center as [NaN, NaN].

Specific issues I had were dragging the map and olView.calculateExtent(olMap.getSize()) returning [NaN,NaN,NaN,NaN]

**Comment by ahocevar** [comment]
Please use the types listed in the API docs to avoid such issues. The argument of `ol.View#setCenter()` is `{Array.<number>}`.

**Comment by TDesjardins** [comment]
Why not checking types with assertions to prevent bad arguments?

**Comment by ahocevar** [comment]
We decided not to do so.

**Comment by derekwatling** [comment]
I am using parseFloat() now that I know what was causing my problem. What was confusing is that it looked like it is working (moved the view center to the specified position) but then I couldn't drag the map.

**Comment by TDesjardins** [comment]
What are the reasons for this decision? Is there new knowledge of good software design? I think you and the users of ol3 could save a lot of time when validating input arguments (type checking included). I suppose there will be a lot of more issues of this kind in future.

Best wishes

**Comment by avbentem** [comment]
In the latest 4.0.1 (and in 3.20.1), using an array of string values (such as `["123", "456"]`, in my case badly parsed from a URL) causes a complete freeze of Chrome, Firefox, Safari and Internet Explorer.

Live example for 4.0.1 on https://codepen.io/anon/pen/jBOwjL

```
var view = new ol.View({
  center: [546000, 6868000],
  zoom: 13
});

var map = new ol.Map({
  target: 'map',
  layers: [
    new ol.layer.Tile({
      source: new ol.source.OSM()
    })
  ],
  view: view
});

// This will freeze the browser
view.setCenter(["546000", "6868000"]);
```

**Comment by ahocevar** [comment]
Please supply numbers, as this is the expected and documented type.

**Comment by blindfish3** [comment]
I just encountered this issue (by inadvertently passing strings from a database request to `View.setCenter()`) and have to say I'm very surprised at the decision not to do any input validation; particularly given the repercussions in this particular case.  If the result was silent failure I could almost accept it; but the symptoms I experienced look like a massive memory leak, crashing the browser and bringing my system to its knees.  That's very uncool.

I'd encourage you to either do type-checking; or, failing that, at least prevent memory leaks when the wrong type is passed...
